### PR TITLE
[BugFix] overflow in parsing parquet int96 timestamp 

### LINKED ISSUE #22355: overflow in loading parquet with int96 timestamp
### Steps to reproduce the behavior (Required)
1. Create table with datetime column
```
CREATE TABLE `tbl_datetime` ( `col_datetime` datetime NULL COMMENT "" ) ENGINE=OLAP DUPLICATE KEY(`col_datetime`) COMMENT "OLAP" DISTRIBUTED BY HASH(`col_datetime`) BUCKETS 1 PROPERTIES ( "replication_num" = "1" ); 
```
3. Load parquet with int96 timestamp
```
LOAD LABEL db0.label3 (     DATA INFILE("hdfs://127.0.0.1:9000/parquet_int96/*")     INTO TABLE tbl_datetime    FORMAT AS "parquet"     (col_datetime)  ) WITH BROKER ;
```
The content of `int96_timestamp.parquet` is
```
0 9999-12-31 23:59:59.999900000
1 2006-01-02 07:04:05.000000000
```
which is generated by script
```
import datetime
import pandas as pd
import pyarrow as pa
from pyarrow import parquet as pq

df = pd.DataFrame({"col_datetime": [datetime.datetime(9999,12,31, 15, 59,59, 9999), datetime.datetime(2006,1,2, 7, 4, 5)]})
a_df = pa.Table.from_pandas(df)
print(a_df.schema) # a: timestamp[us] 

pq.write_table(a_df, "int96_timestamp.parquet", use_deprecated_int96_timestamps=True, version="1.0")

pfile = pq.ParquetFile("int96_timestamp.parquet")
print(pfile.schema_arrow) # a: timestamp[ns]
df1 = pq.read_table("int96_timestamp.parquet").to_pandas()
print(df1)
```

5.  Check records in table
### Expected behavior (Required)
```
+---------------------+
| col_datetime        |
+---------------------+
| 9999-12-31 23:59:59 |
| 2006-01-02 15:04:05 |
+---------------------+
```
### Real behavior (Required)
```
MySQL [db0]> select * from tbl_datetime;
+---------------------+
| col_datetime        |
+---------------------+
| 1816-03-30 06:01:51 |
| 2006-01-02 15:04:05 |
+---------------------+

```
### StarRocks version (Required)
2.2.13,2.3.6,2.4.5, 2.5.4,3.0.0

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/22355 https://github.com/StarRocks/starrocks/issues/14377

## Problem Summary(Required) ：

The parquet format would use int96 to store big datetime like `9999-12-31 23:59:59`, which leads to overflow when it is cast to int64.
The parquet reader provides an option to set the unit of int96 timestamp in https://github.com/apache/arrow/pull/10461.
This PR adds a config `parquet_coerce_int96_timestamp_unit` for BE to set the unit of reading parquet int96 timestamp.
With the default value `MICRO`, the maximum datetime value `9999-12-31 23:59:59.999999` in MySQL could be correctly handled.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: be/src/exec/parquet_reader.cpp [hunk 1]
```diff
@@ -47,6 +48,22 @@ ParquetReaderWrap::ParquetReaderWrap(std::shared_ptr<arrow::io::RandomAccessFile
     _properties.enable_buffered_stream();
     _properties.set_buffer_size(8 * 1024 * 1024);
     _filename = (reinterpret_cast<ParquetChunkFile*>(_parquet.get()))->filename();
+
+    const std::string unit_config = config::parquet_coerce_int96_timestamp_unit;
```
Comment (chaoyli) [inline be/src/exec/parquet_reader.cpp:52]: If not set the parquet_coerce_int96_timestamp_unit, what's the behavior?

Comment (rickif) [reply be/src/exec/parquet_reader.cpp:52]: The config is empty, and the unit would be default value `MICRO`.

File: be/src/common/config.h [hunk 2]
```diff
@@ -948,4 +948,11 @@ CONF_mInt64(load_tablet_timeout_seconds, "30");
 
 CONF_mBool(enable_pk_value_column_zonemap, "true");
 
+// timestamp unit to use for INT96-encoded timestamps in parquet.
+// SECOND, MICRO, MILLI, NANO
+// The default unit is MICRO, which is the precision of DATETIME/TIMESTGAMP in MySQL.
+// https://dev.mysql.com/doc/refman/8.0/en/datetime.html
+// A DATETIME or TIMESTAMP value can include a trailing fractional seconds part in up to microseconds (6 digits) precision
+CONF_mString(parquet_coerce_int96_timestamp_unit, "MICRO");
```
Comment (meegoo) [inline be/src/common/config.h:956]: Is it suitable as a config for BE? any situation we will make it different from each BE?

Comment (rickif) [reply be/src/common/config.h]: The config item is moved to fe.conf.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/FileScanNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -2267,4 +2267,14 @@ public class Config extends ConfigBase {
...
+
+    /*
+     * timestamp unit to use for INT96-encoded timestamps in parquet.
+     * SECOND, MICRO, MILLI, NANO
+     * The default unit is MICRO, which is the precision of DATETIME/TIMESTAMP in MySQL.
+     * https://dev.mysql.com/doc/refman/8.0/en/datetime.html
+     * A DATETIME or TIMESTAMP value can include a trailing fractional seconds part in up to microseconds (6 digits) precision
+     */
+    @ConfField
+    public static String parquet_coerce_int96_timestamp_unit = "MICRO";
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:2279]: If we need to parse the MICRO timestamp, why do we need this configuration?

File: be/src/exec/parquet_reader.cpp [hunk 4]
```diff
@@ -158,6 +164,21 @@ void ParquetReaderWrap::close() {
     [[maybe_unused]] auto st = _parquet->Close();
 }
 
+void ParquetReaderWrap::set_coerce_int96_timestamp_unit(std::string unit) {
+    // The default unit is MICRO, which is the precision of DATETIME/TIMESTAMP in MySQL.
+    if (unit == "SECOND") {
+        _parquet_int96_timestamp_unit = arrow::TimeUnit::SECOND;
+    } else if (unit == "MILLI") {
+        _parquet_int96_timestamp_unit = arrow::TimeUnit::MILLI;
+    } else if (unit == "MICRO") {
```
[the remark this answers is not in the document]
Comment (rickif) [reply be/src/exec/parquet_reader.cpp]: Already removed.

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**